pkg/chunk/schema_config.go

```diff
@@ -148,6 +151,16 @@ func (cfg *SchemaConfig) ForEachAfter(t model.Time, f func(config *PeriodConfig)
 func (cfg PeriodConfig) CreateSchema() (BaseSchema, error) {
 	buckets, bucketsPeriod := cfg.createBucketsFunc()
 
+	// Ensure Index table prefix exists
+	if cfg.IndexTables.Prefix == "" {
+		return nil, errConfigIndexNotSet
+	}
+
+	//Ensure Chunks table prefix exists
+	if cfg.ChunkTables.Prefix == "" {
```

Comment:
I think this is not correct. Even if obscure, as of today having an empty prefix for the chunk tables means that the table manager should not create such tables ([see code here](https://github.com/cortexproject/cortex/blob/311ab3eb147773ba52a07150cdf5511a3797e06b/pkg/chunk/table_manager.go#L415-L419)). This is a valid setting used by people storing chunks on a object storage.

Reply:
@pracucci  I agree, I learned it the hard way. My configuration was missing `chunks` section and all I was getting was `Table name empty error` which was most likely coming from table manager. After comparing  config file, provided by @bboreham I realized that I was missing this piece of information in my config file. So, I thought there should be a check for that as well.

Reply:
So you need to narrow the check. There should only be a check if the chunk store intends to use tables.
Chunks can be stored in non-table store, like S3 or Swift.
This may simplify the changes to tests that you made; I don't think so many should be necessary.

Reply:
Ohh! Okay.

Reply:
ohh! okay. Looking deeper into the code.